Gnosis-Builders/faucet-server
# Blacklist ip wallets

## Explanation

We want to blacklist bots addresses and ipAddress and fail silently when we notice a blacklisted user.

## Manual Testing Steps

1. Set the environment variable `MAX_REQUESTS_BEFORE_BLACKLIST` to say 2
2. Run code locally using `yarn start:dev`
3. On another terminal run `yarn rock` this will generate an https url for you. Note it down.
4. Go to faucet-ui and set the `REACT_APP_BACKEND_URL` to the above https url
5. Start faucet-ui using `yarn start`
6. Make a request to the faucet 2 times
7. Each time, your wallet balance should increase
8. In your browser run `http://localhost:7002/api/run-blacklist-check` - this will be run in production by easycron.com
9. Make a third request to the faucet
10. This time your wallet balance should not increase and there should not errors

NOTE: You can only make one request per day to the faucet. If you're running this locally, you can open the d14n.data sqlite  DB in any sql browser and after every request set `user.expiry` to a past date (long value)

## Pre-merge author checklist
